### PR TITLE
Use case insensitive comparison when checking whether implicit package references are overridden

### LINKED ISSUE #679: CheckForImplicitPackageReferenceOverrides should ignore case when removing duplicates.
Ported from https://github.com/dotnet/roslyn-project-system/issues/1213

**Version:** 2017 int.d15rel 15.0.26116.0

**Repro:**
git clone https://github.com/aspnet/Razor.git
cd Razor
.\build.cmd initialize
open Razor.sln in VS

See https://github.com/dotnet/roslyn-project-system/issues/1213#issuecomment-273576112 for details.

### PR DESCRIPTION
**Pull request by dsplaisted**
Fixes #679

### CODE REVIEW

**Review by srivatsn — APPROVED** [review]

**Comment by dsplaisted** [comment]
@MattGertz for approval

#### Scenario

A user has a `PackageReference` in the project file for a package that is also implicitly added by the SDK, but they use a different capitalization (which is pretty likely to happen when editing project files manually, as the capitalization we use is `NETStandard.Library` and `Microsoft.NETCore.App`).  We are supposed to give them a warning telling them they don't need the explicit reference.  However, we currently don't show the warning if the capitalization doesn't match.  This fixes that.

#### Bug

#679

#### Workarounds

Be careful about capitalization of package IDs, or know via other means that we implicitly add package references for NETStandard.Library and Microsoft.NETCore.App.

#### Risk

Low

#### Performance Impact

Negligible. This adds a conversion to lowercase when checking for duplicate PackageReference items

#### Regression Analysis

Not a regression

File: src/Tasks/Microsoft.NET.Build.Tasks/CheckForImplicitPackageReferenceOverrides.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@ public class CheckForImplicitPackageReferenceOverrides : TaskBase
 
         protected override void ExecuteCore()
         {
-            var duplicateItems = PackageReferenceItems.GroupBy(i => i.ItemSpec).Where(g => g.Count() > 1);
+            var duplicateItems = PackageReferenceItems.GroupBy(i => i.ItemSpec.ToLowerInvariant()).Where(g => g.Count() > 1);
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/CheckForImplicitPackageReferenceOverrides.cs]: Use the GroupBy overload that takes a comparer and pass StringComparer.OrdinalIgnoreCase

**Review by nguerrera — CHANGES_REQUESTED** [review]
See above

**Comment by MattGertz** [comment]
@dsplaisted Can you clarify what the behavior is if we don't fix this?  Drilling into the associated issues, I see the "Crippling" log file reference, but I don't see how that translates to actual user experience.

**Comment by dsplaisted** [comment]
@MattGertz If we don't fix dotnet/roslyn-project-system#1290, then the experience will be a message box will show in Visual Studio similar to https://github.com/dotnet/roslyn-project-system/issues/1290#issuecomment-275273940, but things should work mostly correctly after clicking OK.

If we do fix that bug, then I believe the result will be that the `PackageReference` in the project file will be silently ignored, which would be very confusing to users if they specified a different version than the implicit `PackageReference`.

**Comment by dsplaisted** [comment]
@MattGertz Per [your comment](https://github.com/dotnet/roslyn-project-system/issues/1290#issuecomment-275479806) on the other issue, did you mean to mark this one as approved to bring to shiproom also?

**Comment by MattGertz** [comment]
Yes, thanks, I forgot.

**Comment by 333fred** [comment]
@MattGertz approved to merge, or pending shiproom?

**Comment by MattGertz** [comment]
Thanks for catching.

**Comment by natidea** [comment]
Approved to merge via [375369](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/375369)

**Review by nguerrera — APPROVED** [review]

**Review by natidea — APPROVED** [review]

**Pull request merged**